### PR TITLE
feat/adding-components

### PR DESCRIPTION
**Pull request by Daphne210**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

### CODE REVIEW

File: src/components/data-table/downloadable-table.component.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { useTable } from "react-table";
+import { CSVLink } from "react-csv";
+
+const TableWithCSVDownload = ({ data, columns }) => {
```
Comment (akileng56) [inline src/components/data-table/downloadable-table.component.tsx]: @Daphne210 Is this used any where?

File: src/components/data-table/data-table.resource.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
+export interface FacilityResponse {
```
Comment (akileng56) [inline src/components/data-table/data-table.resource.tsx]: File doesn't export any thing so it be **.ts** file.
Also this looks like a typing file. 
Add to index.ts file [https://github.com/METS-Programme/esm-ugandaemr-commons-lib/pull/1/files#diff-a9ba9cbedd19c9f66d564d2e89912890209b98f0a7ef19187877d2587300e476](https://github.com/METS-Programme/esm-ugandaemr-commons-lib/pull/1/files#diff-a9ba9cbedd19c9f66d564d2e89912890209b98f0a7ef19187877d2587300e476)

File: src/components/data-table/data-tables.scss [hunk 3]
```diff
@@ -0,0 +1,177 @@
+@use '@carbon/styles/scss/colors';
```
Comment (akileng56) [inline src/components/data-table/data-tables.scss]: Maintain the file naming **data-table.scss**

File: src/components/ohri-form-launcher/ohri-form-launcher.component.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { Add } from "@carbon/react/icons";
+import styles from "./launcher-with-intent.scss";
+import { useTranslation } from "react-i18next";
+import { OHRIOverflowMenu } from "../overflow-menu-button/ohri-overflow-menu.component";
+
+export const OHRIFormLauncherWithIntent: React.FC<{
```
Comment (akileng56) [inline src/components/ohri-form-launcher/ohri-form-launcher.component.tsx]: Nothing such be named **ohri**
Rename to **ugandaemr-form-...** or keep it at **form-launcher....**

File: src/utils/createOHRIDashboardLink.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { SideNavLink, SideNavMenu, SideNavMenuItem } from "@carbon/react";
+import { ExtensionSlot, navigate } from "@openmrs/esm-framework";
+import styles from "./sidenav-links.scss";
+
+export const createOHRIDashboardLink = (meta) => {
```
Comment (akileng56) [inline src/utils/createOHRIDashboardLink.tsx]: Nothing OHRI rename

File: src/constants.ts [hunk 6]
```diff
@@ -0,0 +1,271 @@
+export const careSetting = "6f0c9a92-6f24-11e3-af88-005056821db0";
+
```
Comment (akileng56) [inline src/constants.ts:2]: Are those concepts necessary? where are they used

File: src/utils/ohri-sidebar.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export function scrollIntoView(viewId: string) {
```
Comment (akileng56) [inline src/utils/ohri-sidebar.ts]: Rename the file

File: src/workspace/ohri-workspace-utils.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import { getSyncLifecycle } from "@openmrs/esm-framework";
```
Comment (akileng56) [inline src/workspace/ohri-workspace-utils.ts]: Rename file

**Review by akileng56 — CHANGES_REQUESTED** [review]
@Daphne210 Check the comments

File: src/components/data-table/downloadable-table.component.tsx (hunk 1 above)
Comment (Daphne210) [reply src/components/data-table/downloadable-table.component.tsx]: This was deleted. Had gotten the entire component (data-table) from the Data Visualizer

File: src/components/data-table/data-tables.scss (hunk 3 above)
Comment (Daphne210) [reply src/components/data-table/data-tables.scss]: This was adjusted accordingly.

File: src/constants.ts (hunk 6 above)
Comment (Daphne210) [reply src/constants.ts:2]: This was deleted as well as it was not being used anywhere.

File: src/utils/ohri-sidebar.ts (hunk 7 above)
Comment (Daphne210) [reply src/utils/ohri-sidebar.ts]: This was adjusted

File: src/workspace/ohri-workspace-utils.ts (hunk 8 above)
Comment (Daphne210) [reply src/workspace/ohri-workspace-utils.ts]: This was adjusted

File: src/utils/createOHRIDashboardLink.tsx (hunk 5 above)
Comment (Daphne210) [reply src/utils/createOHRIDashboardLink.tsx]: This was adjusted.

File: src/components/ohri-form-launcher/ohri-form-launcher.component.tsx (hunk 4 above)
Comment (Daphne210) [reply src/components/ohri-form-launcher/ohri-form-launcher.component.tsx]: This was adjusted.

File: src/components/data-table/data-table.resource.tsx (hunk 2 above)
Comment (Daphne210) [reply src/components/data-table/data-table.resource.tsx]: This was deleted as well.

**Comment by Daphne210** [comment]
> @Daphne210 Check the comments
The changes have been resolved.

File: src/item-selector/parameters-container.scss [hunk 9]
```diff
@@ -0,0 +1,52 @@
+@use '@carbon/styles/scss/colors';
```
Comment (akileng56) [inline src/item-selector/parameters-container.scss]: Rename file to **item-selector.scss**

**Review by akileng56 — CHANGES_REQUESTED** [review]

File: src/item-selector/parameters-container.scss (hunk 9 above)
Comment (Daphne210) [reply src/item-selector/parameters-container.scss]: This has been resolved

**Pull request merged**